perun-rpc/src/main/java/cz/metacentrum/perun/rpc/methods/MembersManagerMethod.java

```diff
@@ -1884,17 +1885,33 @@ public Void call(ApiCaller ac, Deserializer parms) throws PerunException {
 	 * @param vo vo
 	 * @param query query with page information
 	 * @param attrNames attribute names
+	 * @param searchString (optional) get only members based on this string and specific parameters in DB,
+	 *        if no searchString is provided, get all members
+	 * @param statuses (optional) get only members with status specified here,
+	 *        if no statuses are provided, get members with any status
 	 *
 	 * @return Paginated<RichMember> page of requested rich members
 	 * @throw VoNotExistsException if there is no such vo
 	 */
 	getMembersPage {
 		@Override
 		public Object call(ApiCaller ac, Deserializer parms) throws PerunException {
+
+			List<Status> statuses = new ArrayList<>();
+			if (parms.contains("statuses")) {
+				statuses = parms.readList("statuses", String.class).stream()
+					.map(Status::valueOf)
+					.collect(Collectors.toList());
+			}
+
+			String searchString = parms.contains("searchString") ? parms.readString("searchString") : "";
+
 			return ac.getMembersManager().getMembersPage(ac.getSession(),
 					ac.getVoById(parms.readInt("vo")),
 					parms.read("query", MembersPageQuery.class),
-					parms.readList("attrNames", String.class));
+					parms.readList("attrNames", String.class),
+					searchString,
```

Comment:
Please, don't add new parameters for this. Add these values to the MembersPageQuery.

Reply:
Ok, I've added the new parameters to MembersPageQuery.